<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Windows tutorial &mdash; SPARTA 1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="SPARTA 1.0 documentation" href="index.html" />
    <link rel="next" title="Linux tutorial" href="lintut.html" />
    <link rel="prev" title="Mac OS X tutorial" href="mactut.html" /> 
  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="lintut.html" title="Linux tutorial"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="mactut.html" title="Mac OS X tutorial"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">SPARTA 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="windows-tutorial">
<span id="wintut"></span><h1>Windows tutorial<a class="headerlink" href="#windows-tutorial" title="Permalink to this headline">¶</a></h1>
<p><strong>Download the workflow</strong>: <a class="reference external" href="http://www.github.com/biobenkj/SPARTA_Windows/archive/master.zip">SPARTA for Windows</a></p>
<ol class="arabic simple">
<li><a class="reference internal" href="#introductionwin"><span>Introduction</span></a></li>
<li><a class="reference internal" href="#basic-terminal-commandswin"><span>Basic Terminal Commands</span></a></li>
<li><a class="reference internal" href="#install-dependencieswin"><span>Install Dependencies</span></a></li>
<li><a class="reference internal" href="#initializing-spartawin"><span>Initializing SPARTA</span></a></li>
<li><a class="reference internal" href="#analyzing-example-datawin"><span>Analyzing Example Data</span></a></li>
<li><a class="reference internal" href="#analyzing-your-datawin"><span>Analyzing Your Data</span></a></li>
<li><a class="reference internal" href="#identifying-potential-batch-effectswin"><span>Identifying Potential Batch Effects</span></a></li>
<li><a class="reference internal" href="#altering-workflow-execution-optionswin"><span>Altering Workflow Execution Options</span></a></li>
</ol>
<div class="section" id="introduction">
<span id="introductionwin"></span><h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>Many bioinformatics software packages and workflows require the user to utilize them from
the command line or terminal. SPARTA is no different. The reason the command line interface
is utilized is that a great deal of power and flexibility can be gained without the use of
a graphical user interface (GUI). Further, a GUI can be difficult to implement across various
platforms. To find the command line interface/Terminal on Windows, go to Windows start button -&gt; Search -&gt; Type in: cmd -&gt; Terminal is now open to enter commands.</p>
<img alt="Navigate to terminal on Windows" class="align-center" src="_images/opentermwindows.jpg" />
<p>Decompress the SPARTA_Windows-master.zip file by double-clicking on it. Now, drag and drop the
decompressed folder onto your desktop.</p>
<p>SPARTA expects either compressed (.gz) or uncompressed FASTQ files (.fq or .fastq) as input,
with a reference genome file in FASTA format and a genome feature file (.gtf) within the folder
that contains the input data. To see an example of appropriate input data, look inside the
ExampleData folder within the SPARTA_Windows-master folder.</p>
<p>To download a reference genome and genome feature file for your favorite bacteria, go to
the <a class="reference external" href="http://bacteria.ensembl.org/info/website/ftp/index.html">Ensembl website</a>.</p>
</div>
<div class="section" id="basic-terminal-commands">
<span id="basic-terminal-commandswin"></span><h2>Basic Terminal Commands<a class="headerlink" href="#basic-terminal-commands" title="Permalink to this headline">¶</a></h2>
<p>Let&#8217;s have a look at some basic Terminal commands, we will cover the commands necessary to:</p>
<p><strong>1.</strong> Move through folders</p>
<p><strong>2.</strong> List the contents of a folder</p>
<p><strong>3.</strong> Make new folders</p>
<p><strong>4.</strong> Rename files/folders</p>
<p><strong>5.</strong> Delete files/folders</p>
<table border="1" class="docutils">
<colgroup>
<col width="3%" />
<col width="13%" />
<col width="17%" />
<col width="67%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">&nbsp;</th>
<th class="head">Command</th>
<th class="head">What it does</th>
<th class="head">Examples</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><strong>1.</strong></td>
<td>cd</td>
<td>Change directory/folder</td>
<td>cd ~ (this changes to your home directory); cd .. (this goes back one folder)</td>
</tr>
<tr class="row-odd"><td><strong>2.</strong></td>
<td>dir</td>
<td>List the contents of a folder</td>
<td>dir</td>
</tr>
<tr class="row-even"><td><strong>3.</strong></td>
<td>mkdir</td>
<td>Make a new directory/folder</td>
<td>mkdir NewFolder (this will make a new folder called &#8216;NewFolder&#8217; in your current directory)</td>
</tr>
<tr class="row-odd"><td><strong>4.</strong></td>
<td>move</td>
<td>Rename or move a file from one name to another</td>
<td>move file1 file2 (this will rename/move file1 to file2)</td>
</tr>
<tr class="row-even"><td><strong>5.</strong></td>
<td>rm</td>
<td>Remove a file (rmdir is the command to remove a folder)</td>
<td>rm file1 (remove file1); rmdir folder1 (remove folder1)</td>
</tr>
</tbody>
</table>
<p><strong>Basic Command Prompt Commands</strong>:</p>
<div class="highlight-python"><div class="highlight"><pre>x /? = provides syntax info and complete list of all parameters for x (a command, like “cd”)
cd = change directory
cd .. = move to the parent directory
cd\ = move to the root of current drive
cd x = move to the current\x directory
cd z: = change to the z root directory (as opposed to c:\)
copy x y = copy file x to directory y (Ex: D:\games\galaga.exe C:\programs[\awesome.exe]), [] = optional
copy file con = display file contents in console
copy con file.txt = create text file in the console window, end with ctrl+z (^z or F6)
date = change the date
del = delete/erase
del x = deletes all files/folders fitting x
del . = deletes all files within current directory
del *.* = deletes all files within current directory
dir = display contents of current directory (Ex: dir [c:][\programs]), [] = optional
dir *.txt = list all .txt files in current directory
dir *.? = list all files with extensions one character in length in current directory
dir /w /p *.* = display all contents one screen at a time
dir | more = display all contents one line at a time
dir /? = provides syntax info and complete list of all dir parameters
echo = send command line input to display (by default)
echo sometext &gt;&gt; somefile.txt = append line(s) of text to any file
echo sometext &gt; somefile.txt = overwrites file with sometext
erase = delete/erase
exit = exit the command prompt
filename.txt = opens filename.txt in current directory in Notepad (or default .txt program)
format z: = format z drive [Ex: use to format a disc or flash drive]
mkdir x = make directory x in current directory
move x y = more or rename x to y
q = escapes sequential display of contents (i.e. the more parameter)
rd x = remove/delete directory x if it’s empty
ren x y = rename file x to y
time = change the time
type file = display the contents of the file ‘file’ (displays file contents in console)
type file |more = display the contents one line at a time
</pre></div>
</div>
<p><em>Ref. sheet from: http://blog.simplyadvanced.net/cheat-sheet-for-windows-command-prompt/</em></p>
</div>
<div class="section" id="install-dependencies">
<span id="install-dependencieswin"></span><h2>Install Dependencies<a class="headerlink" href="#install-dependencies" title="Permalink to this headline">¶</a></h2>
<p>The SPARTA workflow requires a few things in order to run: Python, Java, NumPy, and R.
If you already have these installed, great! If you don&#8217;t, let&#8217;s start by downloading the
latest version of <a class="reference external" href="https://www.python.org/downloads/release/python-2710/">Python 2</a>
(see image below). You will want to download and install the red boxed version of Python 2.
Follow the prompts to install Python with the default values.</p>
<img alt="Python download for Windows" class="align-center" src="_images/pythonforwin.jpg" />
<p>Great! Let&#8217;s check and see if Java is already installed on your system. Open up the terminal,
(if you don&#8217;t remember how to do this, head back to the <a class="reference internal" href="#introductionwin"><span>Introduction</span></a>) and type:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">java</span> <span class="o">-</span><span class="n">version</span>
</pre></div>
</div>
<p>If Java is already installed, it will produce some output that looks like this:</p>
<div class="highlight-python"><div class="highlight"><pre>java version &quot;1.8.0_31&quot;
Java(TM) SE Runtime Environment (build 1.8.0_31-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.31-b07, mixed mode)
</pre></div>
</div>
<p>If the output does <em>not</em> look something like this, Java is likely not installed and two of
the tools require Java to function (Trimmomatic and FastQC). Let&#8217;s download and install a
suitable version of <a class="reference external" href="http://www.oracle.com/technetwork/java/javase/downloads/index.html">Java</a>
(see image below). You will want to download and install the red boxed version of Java JRE.
Follow the prompts to install Java.</p>
<a class="reference internal image-reference" href="_images/javadownloadmac.jpg"><img alt="Java JRE download for Mac" class="align-center" src="_images/javadownloadmac.jpg" style="width: 500px; height: 300px;" /></a>
<p>To install the remaining dependencies, SPARTA is distributed with installers for each remaining
piece of software, however, there is an ideal order with which to install them.</p>
<p>Navigate to the SPARTA_Windows-master folder and then into the &#8220;Software_To_Install&#8221; folder.
Inside this folder is a series of executable installers. Double-click and install them in the
following order:</p>
<ol class="arabic simple">
<li>numpy</li>
<li>vcredist</li>
<li>HTSeq</li>
<li>R</li>
<li>gzip</li>
</ol>
<p>Now, there is one remaining batch file called &#8220;add_python_and_R_to_path.bat&#8221;. This will add
the Python, R, and gzip executables to your path so you can run them from the terminal. To
execute this script, right-click on the file and then click on the option called &#8220;Run as administrator&#8221;.
Windows may warn you that this script is unsafe because it is from an unknown developer. Click on the
&#8220;Details&#8221; button and then click on &#8220;Run anyway&#8221;.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">If this script is not run, SPARTA will not function properly.</p>
</div>
<p>Congratulations! You&#8217;ve installed the necessary dependencies to run SPARTA!</p>
</div>
<div class="section" id="initializing-sparta">
<span id="initializing-spartawin"></span><h2>Initializing SPARTA<a class="headerlink" href="#initializing-sparta" title="Permalink to this headline">¶</a></h2>
<p>Once SPARTA is initialized, the workflow will seek to identify that all of the necessary
dependencies are met. If they are not satisfied, a message specific to what is not installed
will appear as output in the terminal window.</p>
<p>To initialize SPARTA, go to the Terminal and navigate to the SPARTA_Windows-master folder on your desktop by typing:</p>
<div class="highlight-python"><div class="highlight"><pre>cd Desktop\SPARTA_Windows-master
</pre></div>
</div>
<p>To start the workflow, type:</p>
<div class="highlight-python"><div class="highlight"><pre>python SPARTA.py
</pre></div>
</div>
<p>This will start the software and check for dependencies.</p>
</div>
<div class="section" id="analyzing-example-data">
<span id="analyzing-example-datawin"></span><h2>Analyzing Example Data<a class="headerlink" href="#analyzing-example-data" title="Permalink to this headline">¶</a></h2>
<p>SPARTA is distributed with some example data. Specifically, it is the first 100,000 reads
of each sample from <a class="reference external" href="http://onlinelibrary.wiley.com/doi/10.1111/mmi.12688/abstract">Baker et al.</a>.</p>
<p>To begin the analysis, navigate into the SPARTA_Mac-master folder and drag and drop the folder
called &#8220;ExampleData&#8221; out onto the desktop.</p>
<p>If you haven&#8217;t already, <a class="reference internal" href="#introductionwin"><span>initialize SPARTA</span></a> from the Terminal.</p>
<p>If all the <a class="reference internal" href="#install-dependencieswin"><span>dependencies</span></a> are met, SPARTA will pause and prompt
the user:</p>
<div class="highlight-python"><div class="highlight"><pre>Is the RNAseq data in a folder on the Desktop? (Y or N):
</pre></div>
</div>
<p>Type:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">Y</span>
</pre></div>
</div>
<p>Hit Enter/Return</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">SPARTA assumes the data is located in a folder on the desktop by default. It is easiest if all future analyses have the data in a folder (WITHOUT SPACES IN THE NAME) on the desktop.</p>
</div>
<p>Now it will prompt the user for the name of the folder:</p>
<div class="highlight-python"><div class="highlight"><pre>What is the name of the folder on the Desktop containing the RNAseq data?:
</pre></div>
</div>
<p>Type:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">ExampleData</span>
</pre></div>
</div>
<p>This is the name of the folder on the desktop that contains the input example data.
Hit Enter/Return.
From here, the software will trim, QC, align, and count transcript abundance for each sample.
All output/analyses are put in a folder that SPARTA generates on the desktop called &#8220;RNAseq_Data&#8221;.
Within this folder are separate folders for each SPARTA run that are denoted by the date (e.g. 2015-06-04).
Within these folders are four more folders that separate each step of the analysis and are
called: 1) QC, 2) Bowtie, 3) HTSeq, and 4) DEanalysis.</p>
<p>Once the trimming, QC, alignment, and counting are complete, SPARTA will again pause and prompt
the user for how many experimental conditions exist within the analysis.</p>
<p>The output at this point will look like this:</p>
<a class="reference internal image-reference" href="_images/conditionoutput.jpg"><img alt="Condition output" class="align-center" src="_images/conditionoutput.jpg" style="width: 500px; height: 300px;" /></a>
<p>At the prompt that says:</p>
<div class="highlight-python"><div class="highlight"><pre>How many conditions are there?:
</pre></div>
</div>
<p>Type:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="mi">4</span>
</pre></div>
</div>
<p>Hit Enter/Return.
There are 4 experimental conditions that we are considering:</p>
<ol class="arabic simple">
<li>Glycerol pH 7.0</li>
<li>Glycerol pH 5.7</li>
<li>Pyruvate pH 7.0</li>
<li>Pyruvate pH 5.7</li>
</ol>
<p>Each condition has 2 replicates. The next prompt will read:</p>
<div class="highlight-python"><div class="highlight"><pre>Enter the relevant file names, based on the names given in &#39;SPARTA has these files&#39;, with the replicates separated by a comma.
As an example, please see the &#39;conditions_input_example.txt&#39; in the DEanalysis folder.
Once you have entered the file names, hit Enter/Return:
</pre></div>
</div>
<p>At this point, we need to do a few things.</p>
<ol class="arabic simple">
<li>Navigate to the SPARTA output folder called RNAseq_Data located on the desktop</li>
<li>Go to the current run folder (will be the last folder listed if sorted by name)</li>
<li>Go into the DEanalysis folder</li>
<li>Open the conditions_input.txt file in a text editor (NOT MICROSOFT WORD) such as TextEdit</li>
</ol>
<p>The number of experimental conditions listed are based on the number entered at the prompt
asking &#8220;How many conditions are there?:&#8221;. Thus, in our case, there are 4. The contents of
the file will look like:</p>
<div class="highlight-python"><div class="highlight"><pre>Reference_Condition_Files:
Experimental_Condition_2_Files:
Experimental_Condition_3_Files:
Experimental_Condition_4_Files:
</pre></div>
</div>
<p>We now need to enter the file names of the replicates in each condition. These are comma-separated
file names that correspond to the output given by SPARTA (denoted with red bracket)</p>
<a class="reference internal image-reference" href="_images/conditionoutputhighlight.jpg"><img alt="Condition output highlighted" class="align-center" src="_images/conditionoutputhighlight.jpg" style="width: 400px; height: 300px;" /></a>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The file names are case-sensitive and must be spelled <em>exactly</em> as listed in the output given by SPARTA</p>
</div>
<p>Thus, when all the file names are inputed, the conditions_input.txt file should look like this:</p>
<div class="highlight-python"><div class="highlight"><pre>Reference_Condition_Files: mapgly7a.sam, mapgly7b.sam
Experimental_Condition_2_Files:mapgly5a.sam, mapgly5b.sam
Experimental_Condition_3_Files:mappyr7a.sam, mappyr7b.sam
Experimental_Condition_4_Files:mappyr5a.sam, mappyr5b.sam
</pre></div>
</div>
<p>Now, save the changes by going to File -&gt; Save.
Go back to the terminal and hit Enter/Return. From here, the workflow will perform the differential
gene expression analysis through edgeR. If a batch effect may be present, the output will attempt to
warn the user of the potential, unintended variable that <em>must</em> be accounted for before drawing
experimental conclusions.</p>
<p>All the differential gene expression output is located in the RNAseq_Data -&gt; date of your current run -&gt; DEanalysis
folder. The file output includes:</p>
<ol class="arabic simple">
<li>Differential gene expression tables</li>
<li>MDS plot (somewhat analogous to a principle component analysis plot) which will show whether your replicates group together and treatment groups separate based on the treatment</li>
<li>BCV plot (biological coefficient of variation) to look at gene level variation between samples</li>
</ol>
<p>Congratulations! You&#8217;ve analyzed RNA-seq data from raw reads to differential gene expression!</p>
</div>
<div class="section" id="analyzing-your-data">
<span id="analyzing-your-datawin"></span><h2>Analyzing Your Data<a class="headerlink" href="#analyzing-your-data" title="Permalink to this headline">¶</a></h2>
<p>If you haven&#8217;t already, we recommend working through the <a class="reference internal" href="#analyzing-example-datawin"><span>example data analysis</span></a>
first before attempting to work through your own data set to familiarize yourself with the
workflow.</p>
<p>As stated in the <a class="reference internal" href="#introductionwin"><span>Introduction</span></a>, SPARTA expects either compressed (.gz) or uncompressed FASTQ files (.fq or .fastq) as input,
with a reference genome file in FASTA format and a genome feature file (.gtf) within the folder
that contains the input data on your desktop. To see an example of appropriate input data, look inside the
ExampleData folder within the SPARTA_Windows-master folder.</p>
<p>Now, to analyze your own data, follow the steps to <a class="reference internal" href="#initializing-spartawin"><span>initialize SPARTA</span></a>,
and start the analysis!</p>
<p>If you would like to tweak the analysis options for a given step/tool, have a look at the
<a class="reference internal" href="#altering-workflow-execution-optionswin"><span>Altering Workflow Execution Options</span></a>.</p>
</div>
<div class="section" id="identifying-potential-batch-effectswin">
<span id="identifying-potential-batch-effects"></span><h2>Identifying Potential Batch Effects<a class="headerlink" href="#identifying-potential-batch-effectswin" title="Permalink to this headline">¶</a></h2>
<p>Batch effects can be a source of variation in RNA-seq data that can confound biological conclusions.
In fact, there have been documented cases of batch effects present in published studies that led
readers to be concerned for the validity of the results.</p>
<p>To quote a previously published paper in <a class="reference external" href="http://www.nature.com/nrg/journal/v11/n10/full/nrg2825.html">Nature Reviews Genetics</a>,
&#8220;Batch effects are sub-groups of measurements that have qualitatively different behaviour across conditions and are unrelated
to the biological or scientific variables in a study. For example, batch effects may occur if a subset of experiments was run on
Monday and another set on Tuesday, if two technicians were responsible for different subsets of the experiments or if two different
lots of reagents, chips or instruments were used.&#8221;</p>
<p>Thus, it is paramount that one address batch effects within their data before drawing biological
conclusions from a specific RNA-seq experiment. To illustrate what a batch effect may look
like within the data, we will utilize several different plots.</p>
<p>This first plot comes from the <a class="reference external" href="http://www.nature.com/nrg/journal/v11/n10/full/nrg2825.html">Nature Reviews Genetics</a>
paper where they examine Affymetrix data from a <a class="reference external" href="http://cancerres.aacrjournals.org/content/64/11/4040.long">published bladder cancer study</a>.
You can quickly see that panels C and D from Figure 1 show that samples from batch 1 (blue)
cluster together based on gene expression and samples from batch 2 (orange) cluster together.</p>
<a class="reference internal image-reference" href="_images/batchexample.jpg"><img alt="Batch effect example" class="align-center" src="_images/batchexample.jpg" style="width: 500px; height: 300px;" /></a>
<p>Within RNA-seq data, using SPARTA and the MDS plot generated by edgeR, another example of
batch effects within a study comparing <em>Mycobacterium tuberculosis</em> treated with a compound, we can clearly
see that the mock-treated samples (DMSO) and compound-treated samples (ETZ) separate based on batch (A vs B)
instead of by treatment. Ideally, we would have the samples group together based on treatment
as opposed to batch.</p>
<a class="reference internal image-reference" href="_images/batcheffect.jpg"><img alt="Batch effect example in RNA-seq data" class="align-center" src="_images/batcheffect.jpg" style="width: 500px; height: 300px;" /></a>
<p>If a potential batch effect is detected in the data set, SPARTA will output a message into
the terminal that says:</p>
<div class="highlight-python"><div class="highlight"><pre>IMPORTANT! YOU MAY HAVE A BATCH EFFECT! PLEASE LOOK AT THE MDS PLOT!
</pre></div>
</div>
<p>If this occurs, have a look at the MDS plot in the RNAseq_Data folder -&gt; date of current run -&gt; DEanalysis folder -&gt; MDSplot.png</p>
<p>From here, you will want to adjust your model to account for the batch effect. Within edgeR, this can be
accomplished through an additive linear model. The documentation for edgeR contains a tutorial on
how to deal with batch effects that can be found <a class="reference external" href="http://bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf">here</a>.</p>
<p>Future implementations of SPARTA will include the ability to adjust for batch effects.</p>
</div>
<div class="section" id="altering-workflow-execution-options">
<span id="altering-workflow-execution-optionswin"></span><h2>Altering Workflow Execution Options<a class="headerlink" href="#altering-workflow-execution-options" title="Permalink to this headline">¶</a></h2>
<p>SPARTA is capable of allowing the user to alter the parameters associated with each analysis
step to be tailored to specific use cases. Below are the different parameters that can be altered
and their usage.</p>
<p>Options:</p>
<div class="highlight-python"><div class="highlight"><pre>-h, --help            show this help message and exit
--cleanup=CLEANUP     Clean up the intermediate files to save space. Default
                      action is to retain the intermediate files. Usage:
                      --cleanup=True
--verbose             Display more output for each step of the analysis.
--noninteractive      Non-interactive mode. This is for running SPARTA
                      without any user input. Assumes data is on the
                      desktop. If this option is specified, you must fill
                      out the configuration file (ConfigFile.txt) with the
                      appropriate experimental conditions in the SPARTA
                      folder.

Trimmomatic options:
  The order the options will be run are: ILLUMINACLIP, LEADING,
  TRAILING, SLIDINGWINDOW, MINLEN

  --clip=ILLUMINACLIP
                      ILLUMINACLIP options. MiSeq &amp; HiSeq usually
                      TruSeq3.fa; GAII usually TruSeq2.fa. Default is
                      ILLUMINACLIP:TruSeq3-SE.fa:2:30:10. Usage:
                      --clip=&lt;adapterseqs&gt;:&lt;seed mismatches&gt;:&lt;palindrome
                      clip threshold&gt;:&lt;simple clip threshold&gt;
  --lead=LEADING      Set the minimun quality required to keep a base.
                      Default is LEADING=3. Usage: --lead=&lt;quality&gt;
  --trail=TRAILING    Set the minimum quality required to keep a base.
                      Default is TRAILING=3. Usage: --trail=&lt;quality&gt;
  --slidewin=SLIDINGWINDOW
                      SLIDINGWINDOW options. Default is SLIDINGWINDOW:4:15.
                      Usage: --slidewin=&lt;window_size&gt;:&lt;required_quality&gt;

HTSeq options:
  --stranded=STRANDED
                      Stranded options: yes, no, reverse. Default is
                      --stranded=reverse. Usage: --stranded=yes/no/reverse
  --order=ORDER       Order options: name, pos. Usage: --order=name/pos.
  --minqual=MINQUAL   Skip all reads with quality lower than the given
                      value. Default is --minqual=10. Usage:
                      --minqual=&lt;value&gt;
  --idattr=IDATTR     Feature ID from the GTF file to identify counts in the
                      output table Default is --idattr=gene_id. Usage:
                      --idattr=&lt;id attribute&gt;
  --mode=MODE         Mode to handle reads overlapping more than one
                      feature. Default is --mode=union. Usage: --mode=union
                      /intersection-strict/intersection-nonempty
</pre></div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Windows tutorial</a><ul>
<li><a class="reference internal" href="#introduction">Introduction</a></li>
<li><a class="reference internal" href="#basic-terminal-commands">Basic Terminal Commands</a></li>
<li><a class="reference internal" href="#install-dependencies">Install Dependencies</a></li>
<li><a class="reference internal" href="#initializing-sparta">Initializing SPARTA</a></li>
<li><a class="reference internal" href="#analyzing-example-data">Analyzing Example Data</a></li>
<li><a class="reference internal" href="#analyzing-your-data">Analyzing Your Data</a></li>
<li><a class="reference internal" href="#identifying-potential-batch-effectswin">Identifying Potential Batch Effects</a></li>
<li><a class="reference internal" href="#altering-workflow-execution-options">Altering Workflow Execution Options</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="mactut.html"
                        title="previous chapter">Mac OS X tutorial</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="lintut.html"
                        title="next chapter">Linux tutorial</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/wintut.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="lintut.html" title="Linux tutorial"
             >next</a> |</li>
        <li class="right" >
          <a href="mactut.html" title="Mac OS X tutorial"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">SPARTA 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &copy; Copyright 2015, Benjamin K Johnson.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3.1.
    </div>
  </body>
</html>